test/samples/if_succeeds_on_contract_inheritance.instrumented.sol

```diff
@@ -21,28 +24,31 @@ contract Foo is __scribble_ReentrancyUtils {
         _v.__scribble_check_invs_at_end = __scribble_out_of_contract;
         __scribble_out_of_contract = false;
         y = _original_Foo_inc(x);
-        if (!(b == (a ^ 1))) {
-            emit AssertionFailed("2: ");
-            assert(false);
-        }
         if (_v.__scribble_check_invs_at_end) __scribble_check_state_invariants();
         __scribble_out_of_contract = _v.__scribble_check_invs_at_end;
     }
 
-    function _original_Foo_inc(uint x) private view returns (uint y) {
+    function _original_Foo_inc(uint x) private pure returns (uint y) {
         return x + 1;
     }
 
+    function inc2(uint x) virtual public returns (uint y) {
+        vars2 memory _v;
+        _v.__scribble_check_invs_at_end = __scribble_out_of_contract;
+        __scribble_out_of_contract = false;
+        y = _original_Foo_inc2(x);
+        if (_v.__scribble_check_invs_at_end) __scribble_check_state_invariants();
+        __scribble_out_of_contract = _v.__scribble_check_invs_at_end;
+    }
+
+    function _original_Foo_inc2(uint x) private pure returns (uint y) {}
+
     /// Check only the current contract's state invariants
     function __scribble_Foo_check_state_invariants_internal() internal {
```

Comment:
We don't want to emit a `__scribble_Foo_check_state_invariants_internal` function for a contract that doesn't have any `invariant` contract annotations. To fix this we need to update the logic in `computeContractsNeedingInstr` in scribble.ts

Reply:
See https://github.com/ConsenSys/scribble/pull/29 for a suggested fix.